dftimewolf/lib/exporters/gce_disk_export_dd.py

```diff
@@ -0,0 +1,202 @@
+          and remote_instance_name is provided it will select the instance's
+          boot disk.
+      boot_image_project: Name of the project where the boot disk image is
+          stored.
+      boot_image_family: Name of the image to use to create the boot disk.
+    """
+    self.source_project = gcp_project.GoogleCloudProject(source_project_name)
+    self.remote_instance_name = remote_instance_name
+    self.source_disk_names = []
+    if source_disk_names:
+      self.source_disk_names = source_disk_names.split(',')
+    self.all_disks = all_disks
+
+    self.source_disks = self._FindDisksToCopy()
+    if not self._VerifyDisksInSameZone(self.source_disks):
+      self.ModuleError('All disks need to be in the same Zone.', critical=True)
+    self._DetachDisks(self.source_disks)
+    # Add a trailing slash if it's not already there.
+    self.gcs_output_location = os.path.join(gcs_output_location, '')
+    self.startup_script = self._ReadExportScript(_EXPORT_STARTUP_SCRIPT)
+    self.boot_image_project = boot_image_project
+    self.boot_image_family = boot_image_family
+
+  def Process(self) -> None:
+    """Creates and exports disk image to the output bucket."""
+    export_instance_name = gcp_common.GenerateUniqueInstanceName(
+      'dftimewolf-export')
+    self.startup_script = self.startup_script.format(
+      project_id=self.source_project.project_id,
+      archive_bucket=self.gcs_output_location,
+      zone=self.source_disks[0].zone,
+      instance_name=export_instance_name)
+    export_instance = self.source_project.compute.CreateInstanceFromArguments(
+      instance_name=export_instance_name,
+      machine_type='{0:s}-{1:d}'.format('e2-standard', 16),
+      zone=self.source_disks[0].zone,
+      boot_disk_type='pd-standard',
+      boot_disk_size=20,
+      boot_image_project=self.boot_image_project,
+      boot_image_family=self.boot_image_family,
+      metadata={'startup-script': self.startup_script},
+      data_disks=self.source_disks,
+      additional_scopes=['https://www.googleapis.com/auth/cloud-platform'])
+
+    while not self._ExportJobFinished():
+      self.logger.info(
+        ('Waiting for export instance {0:s} to finish exporting disks. '
+        'All non-boot disk must have "archive" and "archive_hash_verified" '
+        'labels equal "true" for the export job to finish.').format(
+          export_instance.name))
+      time.sleep(30)
+
+    export_instace_api_object = export_instance.GetOperation()
+    for disk in self.source_disks:
+      for key_value_dict in export_instace_api_object.get(
+        'metadata', {}).get('items', []):
+        if not key_value_dict['key'].startswith('archive_path_'):
+          continue
+        archived_disk = key_value_dict['key'].partition('archive_path_')[-1]
+        if archived_disk == disk.name:
+          value_tuple = key_value_dict['value'].partition(r'\n')
+          disk_path_gcs = value_tuple[-1]
```

Comment:
Did you mean value_tuple[1] here, or is `-1` what you expect? I guess for a tuple -1 == 1, but just making sure this is WAI

Reply:
Good catch, I'd use 1 instead of -1 if it's indeed a tuple.

Reply:
I remember the metadata comes sometimes prefixed with some characters (new line), and value_tuple[1] sometimes have 'archive_path_' itself and not the partition after, -1 always worked

Reply:
In that case I would add a comment clarifying that we're explicitly reading the last value of the returned result, which is inconsistent in size across calls

Reply:
Done